synapse/lib/datapath.py

```diff
@@ -43,9 +80,10 @@ def step(self, path):
         '''
         Step to the given DataElem within the tree.
         '''
+        if isinstance(path, str):
```

Comment:
It may be best if this API was updated to assume path is a DataPath, and plumb a higher function that's stepByStr or something.

Reply:
Per discussion in person, the only API will accept a DataPath object.